### PR TITLE
Fixed time handling for model exchange

### PR DESCRIPTION
**Pull request by lochel**

### CODE REVIEW

File: src/OMSimulatorLib/SystemSC.cpp [hunk 1]
```diff
@@ -806,15 +833,27 @@ oms_status_enu_t oms::SystemSC::doStepCVODE()
...
-      if (fmi2OK != fmistatus) logError_FMUCall("fmi2_setTime", fmus[i]);
     }
 
     if (flag == CV_ROOT_RETURN || time == tnext)
     {
-      //logInfo("event found!!! " + std::to_string(time));
+      logDebug("event found!!! " + std::to_string(time));
+
+      // set time
+      for (const auto& component : getComponents())
```
Comment (bilderbuchi) [inline src/OMSimulatorLib/SystemSC.cpp:843]: Aren't you concerned about duplicating the same code block 4 times and then having to successfully maintain this/fix bugs in this section, in the future? As opposed to, say, having this snippet in a small function called in 4 places?

Comment (lochel) [reply src/OMSimulatorLib/SystemSC.cpp:843]: I completely agree! 😊

Comment (lochel) [reply src/OMSimulatorLib/SystemSC.cpp:843]: @bilderbuchi feel free to comment on #1386

Comment (bilderbuchi) [reply src/OMSimulatorLib/SystemSC.cpp:843]: ❤️ great!

**Pull request merged**